parlai/core/agents.py

```diff
@@ -154,6 +156,78 @@ def shutdown(self):
         """
         pass
 
+    def self_observe(self, self_message: Message) -> None:
+        """
+        Observe one's own utterance.
+
+        :param self_message:
+            The message corresponding to the output from batch_act.
+        """
+        pass
+
+    def respond(
+        self, text_or_message: Union[str, Message], **other_message_fields
+    ) -> str:
```

Comment:
Should the return value here be `str` or `Message`? 

(I don't have full context as to what @stephenroller was thinking in terms of exactly how this would be used, but want to check to see if the `str` return in the original API suggestion is intentional or a fluke.)

Reply:
The goal of this is to build a utility function that goes str->str to make the simplest case boilerplate free -- many people just want to load BB and get the response. More complex cases can still be manually observed/act'd.

Reply:
@moyapchen Can you take a look at the issue? Stephen has fleshed it out there. https://github.com/facebookresearch/ParlAI/issues/3408